grc/core/blocks/block.py

```diff
@@ -407,7 +407,7 @@ def make_callback(callback):
 
         return [make_callback(c) for c in self.cpp_templates.render('callbacks')]
 
-    def decide_type(self):
+    def format_expr(self, py_type):
         """
         Evaluate the value of the variable block and decide its type.
 
```

Comment:
Modified this method to take the parameter type as an argument. This way the rvalue expression can be formatted appropriately. 

i.e. given the following flowgraph parameters:
- foo = 2
- bar = 3.5
- foobar = foo*bar 

Python generated code will have a variable foobar of type float initaliazed to 7.0. 
However c++ generated code will have the following:
std::string foobar = "foo*bar";

With the changes, c++ generated code now looks like this:
int foo = 2;
double bar = 3/5;
double foobar = foo*bar;